src/main/java/com/example/bookshop/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.example.bookshop.repository.impl;
+
+import com.example.bookshop.exception.DataProcessingException;
+import com.example.bookshop.model.Book;
+import com.example.bookshop.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Repository;
+import java.util.List;
+
+@AllArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public Book save(Book book) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.persist(book);
+            transaction.commit();
+            return book;
+        } catch (RuntimeException e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save book to DB, cause: " + e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+          return entityManager.createQuery("SELECT b FROM Book b", Book.class).getResultList();

```

Comment:
```suggestion
          return entityManager.createQuery("FROM Book", Book.class).getResultList();
```
Don’t  complicate query